code/modules/arcade/mob_hunt/game_server.dm

```diff
@@ -0,0 +1,106 @@
+		reset_cooldown--
+	if(server_status == 0)
+		return
+	client_mob_update()
+	if(normal_spawns.len < max_normal_spawns)
+		spawn_mob()
+
+//leaving this here in case admins want to use it for a random mini-event or something
+/datum/controller/process/mob_hunt/proc/server_crash(recover_time = 3000)
+	server_status = 0
+	for(var/datum/data/pda/app/mob_hunter_game/client in connected_clients)
+		client.disconnect("Server Crash")
+	for(var/obj/effect/nanomob/N in trap_spawns)
+		N.despawn()
+	for(var/obj/effect/nanomob/N in normal_spawns)
+		N.despawn()
+	//just in case
+	normal_spawns.Cut()
+	trap_spawns.Cut()
+	connected_clients.Cut()
+	if(!isnum(recover_time))
+		recover_time = 3000
+	if(recover_time > 0)	//when provided with a negative or zero valued recover_time argument, the server won't auto-restart but can be manually rebooted still
+		//set a timer to automatically recover after recover_time has passed (can be manually restarted if you get impatient too)
+		addtimer(src, "auto_recover", recover_time, TRUE)
+
+/datum/controller/process/mob_hunt/proc/client_mob_update()
+	var/list/ex_players = list()
+	for(var/datum/data/pda/app/mob_hunter_game/client in connected_clients)
+		var/mob/living/carbon/human/H = client.get_player()
+		if(connected_clients[client])
+			if(!H || H != connected_clients[client])
+				ex_players |= connected_clients[client]
+		connected_clients[client] = H
+	if(ex_players.len)	//to make sure we don't do this if we didn't lose any player since the last update
+		for(var/obj/effect/nanomob/N in (normal_spawns + trap_spawns))
+			N.conceal(ex_players)
+
+/datum/controller/process/mob_hunt/proc/auto_recover()
+	if(server_status != 0)
+		return
+	server_status = 1
+	while(normal_spawns.len < max_normal_spawns)		//repopulate the server's spawns completely if we auto-recover from crash
+		spawn_mob()
+
+/datum/controller/process/mob_hunt/proc/manual_reboot()
+	if(server_status && reset_cooldown)
+		return 0
+	for(var/obj/effect/nanomob/N in trap_spawns)
+		N.despawn()
+	for(var/obj/effect/nanomob/N in normal_spawns)
+		N.despawn()
+	server_status = 1
+	reset_cooldown = 25		//25 controller cycle cooldown for manual restarts
+	return 1
+
+/datum/controller/process/mob_hunt/proc/spawn_mob()
+	var/list/nanomob_types = subtypesof(/datum/mob_hunt)
+	var/datum/mob_hunt/mob_info = pick(nanomob_types)
+	new mob_info()
+
+/datum/controller/process/mob_hunt/proc/register_spawn(datum/mob_hunt/mob_info)
+	if(!mob_info)
+		return 0
+	var/obj/effect/nanomob/new_mob = new /obj/effect/nanomob(mob_info.spawn_point, mob_info)
+	normal_spawns += new_mob
+	new_mob.reveal()
+	return 1
+
+/datum/controller/process/mob_hunt/proc/register_trap(datum/mob_hunt/mob_info)
+	if(!mob_info)
+		return 0
+	if(!mob_info.is_trap)
+		return register_spawn(mob_info)
+	var/obj/effect/nanomob/new_mob = new /obj/effect/nanomob(mob_info.spawn_point, mob_info)
```

Comment:
Pooling is actually really bad, performance wise---I've profiled it multiple times, even with updates and optimizations from TG--unfortunately, pooling just plain results in a net loss of performance---even with the changes to vars and how they're accessed in BYOND 511, pooling is still inferior to just creating new instances and GC'ing it when you're done with it.